src/NuGet.Core/NuGet.Common/ExceptionUtilities.cs

```diff
@@ -14,20 +14,42 @@ namespace NuGet.Common
     /// </summary>
     public static class ExceptionUtilities
     {
-        public static void HandleException(Exception ex, ILogger logger)
+        /// <summary>
+        /// Log an exception to an ILogger.
+        /// This will log using NU1000 if the exception does not contain a code.
+        /// </summary>
+        public static void LogException(Exception ex, ILogger logger)
         {
+            LogException(ex, logger, logStackAsError: false);
+        }
+
+        /// <summary>
+        /// Log an exception to an ILogger.
+        /// This will log using NU1000 if the exception does not contain a code.
+        /// </summary>
+        public static void LogException(Exception ex, ILogger logger, bool logStackAsError)
+        {
+            // Unwrap aggregate exceptions.
+            var unwrappedException = Unwrap(ex);
+
             // Log the error
-            if (ExceptionLogger.Instance.ShowStack)
+            var logExceptionMessage = unwrappedException as ILogMessageException;
+            if (logExceptionMessage != null)
             {
-                logger.LogError(ex.ToString());
+                // Log the log message itself.
+                var logMessage = logExceptionMessage.AsLogMessage();
+                logger.Log(logMessage);
             }
             else
             {
-                logger.LogError(ExceptionUtilities.DisplayMessage(ex));
+                // Create a string from the exception.
+                logger.Log(LogMessage.CreateError(NuGetLogCode.NU1000, DisplayMessage(unwrappedException)));
```

Comment:
we should consider improving this in the future. Else everything will throw NU1000 with an unexpected exception.